internal/transformations/js_decode.go

```diff
@@ -10,13 +10,14 @@ import (
 	utils "github.com/corazawaf/coraza/v3/internal/strings"
 )
 
-func jsDecode(data string) (string, error) {
+func jsDecode(data string) (string, bool, error) {
 	if i := strings.IndexByte(data, '\\'); i != -1 {
 		// TODO: This will transform even if the backslash isn't followed by an escape,
 		// but keep it simple for now.
-		return doJsDecode(data, i), nil
+		transformedData := doJsDecode(data, i)
+		return transformedData, data != transformedData, nil
```

Comment:
I think specially here it is easier to track a change rather than compare the data byte by byte.